UnitTest/TestClientAndServer.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Scarlet.Communications;
+using Scarlet.Utilities;
+
+namespace UnitTest
+{
+    [TestClass]
+    public class UnitTestClientAndServer
+    {
+        private static String ReceivedMessage, ReceivedMessage2;
+
+        private void InitializeConnection()
+        {
+            Server.Start(2000, 2001);
+            System.Threading.Thread.Sleep(500);
+            Client.Start("127.0.0.1", 2000, 2001, "TestClient");
+            System.Threading.Thread.Sleep(500);
+        }
+
+        private void InitializeConnectionWithQueueBuffer()
+        {
+            Server.Start(2000, 2001, UsePriorityQueue: false);
+            System.Threading.Thread.Sleep(500);
+            Client.Start("127.0.0.1", 2000, 2001, "TestClient", UsePriorityQueue: false);
+            System.Threading.Thread.Sleep(500);
+        }
+
+        public void MessageHandler(Packet Packet)
+        {
+            ReceivedMessage = UtilData.ToString(Packet.Data.Payload);
+        }
+
+        public void MessageHandler2(Packet Packet)
+        {
+            ReceivedMessage2 = UtilData.ToString(Packet.Data.Payload);
+        }
+
+        [TestMethod]
+        public void BasicTest()
```

Comment:
This is pretty cool. I've never done unit testing before myself, so this will also serve as a good example of how it works. Thanks!